src/Http/Requests/Bulk/DestroyThreads.php

```diff
@@ -33,29 +33,44 @@ public function authorizeValidated(): bool
 
     public function fulfill()
     {
-        $threads = $this->threads();
+        $threads = $this->threads()->get();
+
+        if ($threads->count() === 0) return 0;
+        
+        // Avoid using Eloquent to prevent automatic touching of updated_at
+        $query = \DB::table((new Thread)->getTable())
+            ->whereNull('deleted_at')
+            ->whereIn('id', array_unique($this->validated()['threads']));
 
-        if (config('forum.general.soft_deletes') && isset($this->validated()['permadelete']) && $this->validated()['permadelete'] && method_exists(Thread::class, 'forceDelete'))
+        if (config('forum.general.soft_deletes') && isset($this->validated()['permadelete']) && $this->validated()['permadelete'])
         {
-            $threads->forceDelete();
+            $query->delete();
         }
         else
         {
-            $threads->delete();
+            $query->update(['deleted_at' => \DB::raw('NOW()')]);
         }
 
-        $threadsByCategory = $threads->select('category_id')->distinct()->get();
-        foreach ($threadsByCategory as $thread)
+        $threadsByCategory = $threads->groupBy('category_id');
+        foreach ($threadsByCategory as $threads)
         {
-            $thread->category->syncCurrentThreads();
+            $threadCount = $threads->count();
+            $postCount = $threads->sum('reply_count') + $threadCount; // count the first post of each thread
+            $category = $threads->first()->category;
+
+            $category->update([
+                'newest_thread_id' => $category->getNewestThreadId(),
+                'latest_active_thread_id' => $category->getLatestActiveThreadId(),
+                'thread_count' => \DB::raw("thread_count - {$threadCount}"),
+                'post_count' => \DB::raw("post_count - {$postCount}")
+            ]);
         }
 
-        return $threads->get();
+        return $threads->count();
```

Comment:
Is this correct, as won't it potentially return the count whether or not they were deleted? Wouldn't you want the count returned by the query?

Reply:
The count should be the same because the original query is already scoped to threads that are eligible for deletion, but it wouldn't hurt to go by the delete query instead. I'll try it later.